nym-vpn-apple/Services/Sources/Services/ConfigurationManager/ConfigurationManager.swift

```diff
@@ -6,12 +6,21 @@ import Logging
 public final class ConfigurationManager {
     private let appSettings: AppSettings
     private let logger = Logger(label: "Configuration Manager")
+    private let fallbackEnv = Env.mainnet
 
     // Source of truth in AppSettings.
     // We need to set same settings in tunnel extension as well.
-    private var currentEnv: Env = .mainnet {
-        didSet {
-            appSettings.currentEnv = currentEnv.rawValue
+    // fallbackEnv edge case, when we cannot parse from AppSettings.
+    private var currentEnv: Env {
+        get {
+            guard let env = Env(rawValue: appSettings.currentEnv)
```

Comment:
This looks complex, use null-coalescing instead: `Env(rawValue: appSettings.currentEnv) ?? self.fallbackEnv`